### PR TITLE
DA-90 Request issues from JIRA

### PR DESCRIPTION
**Pull request by michelleshu**
- [x] Ben
- [x] Christian

### CODE REVIEW

File: lib/clAdapters/jira/index.js [hunk 1]
```diff
@@ -56,11 +61,61 @@ export default class JiraAdapter extends Adapter {
...
+
+    return issues;
   }
 
   async runConnectionTest() {
     return await this.makeRequest('myself');
   }
+
+  async getIssueHierarchy() {
+    return await this.makeRequest('issue/createmeta');
```
Comment (bsouthga) [inline lib/clAdapters/jira/index.js]: minor nitpick, but for functions like this when you are immediately awaiting a promise, you might as well just write it as 

```javascript
getIssueHierarchy() {
  return this.makeRequest('issue/createmeta');
}
```

Which is has the same result, but avoids the overhead of transpiling the async function to a generator function, and calling the generator function

File: lib/clAdapters/jira/index.js [hunk 2]
```diff
@@ -56,11 +61,61 @@ export default class JiraAdapter extends Adapter {
...
+  }
+
+  async getUnresolvedEpicsForProject(projectId) {
+    return await this.getAllIssues({
```
Comment (bsouthga) [inline lib/clAdapters/jira/index.js]: same as above for the other methods

**Comment by bsouthga** [comment]
Also, for rate limiting on the app side, you could monkey patch the `makeRequest` function (if that's where all api requests come from) like so...

```javascript
// rate limit to 1 / (5 seconds)
JiraAdapter.prototype.makeRequest =  rateLimit(JiraAdapter.prototype.makeRequest, 5000);
```

**Comment by bsouthga** [comment]
or we could pull the rate limiting stuff out of the app code if we want to use it in here

**Comment by michelleshu** [comment]
@bsouthga The makeRequest adjustment in the app code that you suggested would work perfectly this JIRA code, since all the requests will go through there. However, I can also see the case for a general rate limiting function in this project.

Do you think the rate limiting function should be used for the other adapters?

**Comment by yangchristian** [comment]
Nothing in addition to @bsouthga's comments. Looks good to me.

**Comment by bsouthga** [comment]
I'd say ship it, we can leave the "rate limiting in this repo" discussion for a later date.

**Pull request merged**